extensions/ql-vscode/scripts/source-map.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * This scripts helps finding the original source file and line number for a
+ * given file and line number in the compiled extension. It currently only
+ * works with released extensions.
+ *
+ * Usage: npx ts-node scripts/source-map.ts <version-number> <filename>:<line>:<column>
+ */
+
+import { spawnSync } from "child_process";
+import { basename, resolve } from "path";
+import { pathExists, readJSON } from "fs-extra";
+import { SourceMapConsumer } from "source-map";
+
+if (process.argv.length !== 4) {
+  console.error(
+    "Expected 2 arguments - the version number and the filename:line number",
+  );
+}
+
+const versionNumber = process.argv[2].startsWith("v")
+  ? process.argv[2]
+  : `v${process.argv[2]}`;
+const filenameAndLine = process.argv[3];
+
+async function extractSourceMap() {
+  const sourceMapsDirectory = resolve(
+    __dirname,
+    "..",
+    "artifacts",
+    "source-maps",
+    versionNumber,
+  );
+
+  if (!(await pathExists(sourceMapsDirectory))) {
+    console.log("Downloading source maps...");
+
+    const workflowRuns = runGhJSON<WorkflowRunListItem[]>([
```

Comment:
So, this will only work for as long as the workflow runs are still available from actions. Ideally, we can publish the source map as part of the release process and download the release artifact instead.

Reply:
Definitely! We'd probably need to create a ZIP file and upload it as a release asset. I think we can do this in a follow-up PR.